warn/scrapers/ga.py

```diff
@@ -1,121 +1,108 @@
+        for year in self.years:
+            source = CachedURL(
+                f"https://www.dol.state.ga.us/public/es/warn/searchwarns/list?geoArea=9&year={year}&step=search"
+            )
+            source.cache_dir = self.source.cache_dir
+            yield NoticeList(source=source)
 
-    area = 9  # statewide
 
-    current_year = datetime.now().year
-    first_year = 2002  # first available year
+class HtmlTableRow:
+    """A row in a table."""
 
-    years = list(range(first_year, current_year + 1))
-    years.reverse()
+    def __init__(self, item):
+        """Initialize a new instance."""
+        self.item = item
 
-    # Loop through the years and scrape them one by one
-    output_rows = []
-    for i, year in enumerate(years):
-        # Concoct the URL
-        url = f"{base_url}?geoArea={area}&year={year}&step=search"
-        cache_key = f"{state_code}/{year}.html"
-
-        # Read from cache if available and not this year or the year before
-        if cache.exists(cache_key) and year < current_year - 1:
-            html = cache.read(cache_key)
-        else:
-            # Otherwise, go request it
-            page = utils.get_url(url)
-            html = page.text
-            cache.write(cache_key, html)
-
-        # Scrape out the table
-        new_rows = _parse_table(
-            html,
-            "emplrList",
-            include_headers=i == 0,  # After the first loop, we can skip the headers
-        )
-
-        # Concatenate the rows
-        output_rows.extend(new_rows)
+    def is_header(self):
+        """Test whether the row is a header."""
+        return self.item[0].tag == "th"
 
-    # Write out the results
-    data_path = data_dir / f"{state_code}.csv"
-    utils.write_rows_to_csv(data_path, output_rows)
+    def get_text(self):
+        """Get the text of the row."""
+        return tuple(utils.clean_text(cell.text_content()) for cell in self.item)
 
-    # Return the path to the CSV
-    return data_path
 
+class HmlTable(HtmlListPage):
+    """A table of HTML elements."""
 
-def _parse_table(html, id, include_headers=True):
-    """
-    Parse HTML table with given ID.
+    selector = CSS("table tr")
 
-    Keyword arguments:
-    html -- the HTML to parse
-    id -- the ID of the table to parse
-    include_headers -- whether to include the headers in the output (default True)
+    def process_item(self, item):
+        """Process an item."""
+        row = HtmlTableRow(item)
 
-    Returns: a list of rows
-    """
-    # Parse out data table
-    soup = BeautifulSoup(html, "html.parser")
-    table_list = soup.find_all(id=id)  # output is list-type
+        # Skip the header row, saving headers as property
+        if row.is_header():
+            self.headers = row
+            raise SkipItem("headers")
 
-    # We expect the first table to be there with our data
-    assert len(table_list) > 0
-    table = table_list[0]
+        return dict(zip(self.headers.get_text(), row.get_text()))
```

Comment:
Should the return value be a dict? Or something else? Is `get_text()` the right method name?